addons/main/functions/anomalies/fn_activateTeleport.sqf

```diff
@@ -77,7 +76,18 @@ _proxy = _exit getVariable QGVAR(sound);
 
             if (_doTeleport) then {
                 // [_obj, [((_exitPos select 0) + (random 4) - 2), ((_exitPos select 1) + (random 4) - 2), (_exitPos select 2) ]] remoteExec ["setPos", _obj];
+                
+                try {
+                    [_trg, _exit, _obj] call _onEnterCode;
+                } catch {
+                    hintC ("OnEnter code failed to execute. Affected anomaly at " + str (_pos));
+                };
                 _obj setPos [((_exitPos select 0) + (random 4) - 2), ((_exitPos select 1) + (random 4) - 2), (_exitPos select 2) ];
+                try {
+                    [_trg, _exit, _obj] call _onExitCode;
+                } catch {
+                    hintC ("OnExit code failed to execute. Affected anomaly at " + str (_pos));
+                };
```

Comment:
Not really a fan of try catch in sqf. Other languages are fine, as they catch actually exceptions, sqf does not.

Here instead of the enter and exit codes why not raise local CBA events?

Reply:
I decided that since everything is running on the server, that the code hooks should be server side only too. If you would prefer the code be client side, then I would use CBA events. But in the current implementation, using events is just adding an extra step to calling the code. 

Also, I do not know if raising an event on a local machine is synchronous or not, so I do not know if CBA events would ruin the order of execution. OnEnter is ran before you teleport ("when you enter the teleporter"), you are teleported, and then onExit is ran after you teleport ("when you exit the teleporter").

Try Catch is there to guarantee that bad code does not cause the entire function stop if a player passes bad code. I am not worried about why the code failed to execute; that is on the player to figure out.

That said, if you are fine with it erring out and requiring players to fix it, that is entirely valid to and I will remove the try-catch. But the last bit of code (lines 99-103, timing out the cooldown) will not run in case of an error with onEnter/Exit, which I think causes the teleporter to completely break.

Let me know what you would like or make any changes you prefer.

Reply:
The code here is already running on the sever, raising a local event will only raise on the server itself. A client will not receive it at all. In this case a client really does not need the event, as the whole teleport handling is done on the server. It is up to the mission maker to run code on the client if they want to, we just provide the API that something has occurred.

Mission makers can add the event listener to the server and implement their own handling. Error handling is done by CBA then, the teleport code should still continue to run while the mission maker or server receives the script error that happened in the executed custom code. Same mentality as CBA/ACE, if bad custom code is used, let the code creator fix it, it is a Sisyphean endeavor trying to prevent or correcting user errors.

It is way less handling on our end to do then, certainly no need to compile the code from a string or add a try and catch in case bad code has been given.

Plus easier to document on my end in the wiki as well.